cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/Filters/FilterCollection/FilterConditionInput/index.tsx

```diff
@@ -46,21 +46,18 @@ enum FilterConditionProperty {
   Value = 'value',
 }
 
-interface IFilterConditionInputProps extends WithStyles<typeof styles>, ICreateContext {
+interface IFilterConditionInputProps extends WithStyles<typeof styles> {
   filterID: string;
 }
 
-const FilterConditionInputview: React.FC<IFilterConditionInputProps> = ({
-  classes,
-  filterID,
-  filterToCondition,
-  setFilterToCondition,
-  widgetInfo,
-}) => {
+const FilterConditionInputview: React.FC<IFilterConditionInputProps> = ({ classes, filterID }) => {
+  const { widgetInfo } = useWidgetState();
+  const { filterToCondition, setFilterToCondition } = useFilterState();
+
   const existingCondition = filterToCondition.get(filterID);
 
   const [conditionMode, setConditionMode] = React.useState(
-    existingCondition.get('expression')
+    !isNil(existingCondition.get('expression'))
```

Comment:
It might be more succinct/readable to use existingCondition.has('expression') here

Reply:
Good catch! Resolved